packager/packager.cc

```diff
@@ -2055,26 +2055,36 @@ class PackageInfoFormatter final {
                          const UnorderedMap<core::packages::MangledName, PackageFiles> &packageFiles)
         : gs(gs), packageFiles(packageFiles) {}
 
-    void operator()(std::string *out, core::packages::MangledName mangledName) const {
-        const auto &pkg = gs.packageDB().getPackageInfo(mangledName);
-        out->append("{");
-        out->append("\"name\":");
-        fmt::format_to(back_inserter(*out), "\"{}\",",
-                       absl::StrJoin(pkg.fullName(), "::", core::packages::NameFormatter(gs)));
-        out->append("\"imports\":[");
-        fmt::format_to(back_inserter(*out), "{}", absl::StrJoin(pkg.imports(), ",", ImportFormatter(gs)));
-        out->append("],\"testImports\":[");
-        fmt::format_to(back_inserter(*out), "{}", absl::StrJoin(pkg.testImports(), ",", ImportFormatter(gs)));
-        out->append("],\"files\":[");
-        const auto it = packageFiles.find(mangledName);
+    std::string files(const UnorderedMap<core::packages::MangledName, PackageFiles>::const_iterator it) const {
+        std::string res;
         if (it != packageFiles.end()) {
-            fmt::format_to(back_inserter(*out), "{}", absl::StrJoin(it->second.files, ",", FileListFormatter(gs)));
+            res = absl::StrJoin(it->second.files, ",", FileListFormatter(gs));
         }
-        out->append("], \"testFiles\":[");
+        return res;
+    }
+
+    std::string testFiles(const UnorderedMap<core::packages::MangledName, PackageFiles>::const_iterator it) const {
+        std::string res;
         if (it != packageFiles.end()) {
-            fmt::format_to(back_inserter(*out), "{}", absl::StrJoin(it->second.testFiles, ",", FileListFormatter(gs)));
+            res = absl::StrJoin(it->second.testFiles, ",", FileListFormatter(gs));
         }
-        out->append("]}");
+        return res;
+    }
+
+    void operator()(std::string *out, core::packages::MangledName mangledName) const {
+        const auto &pkg = gs.packageDB().getPackageInfo(mangledName);
+        const auto it = packageFiles.find(mangledName);
+        // clang-format off
+        absl::StrAppend(
+            out,
+            "{\"name\":\"", absl::StrJoin(pkg.fullName(), "::", core::packages::NameFormatter(gs)),
+            "\",\"imports\":[", absl::StrJoin(pkg.imports(), ",", ImportFormatter(gs)),
+            "],\"testImports\":[" , absl::StrJoin(pkg.testImports(), ",", ImportFormatter(gs)),
+            "],\"files\":[", this->files(it),
+            "], \"testFiles\":[", this->testFiles(it),
+            "]}"
+        );
```

Comment:
Personally, I would feel better if we used rapidjson to format the JSON instead of string interpolation, which ensures that special characters are escaped properly. You can see an example in `tracing.cc`.

Reply:
Yeah, that's fair. I'll rework this to use rapidjson instead.